### DISCUSSION TITLE
About search file under codebase with Remote Index Server

### DISCUSSION THREAD

**Discussion by simon8233** [clangd/clangd#1031] [Q&A] [answered]
I am new on VSCode with Clangd Users. 

I deployed one remote index server to serve internal users.

The user tries to search files in ten thousand files codebase. 

it seems no performance gain from the remote index server.

is it expected ? or shall I read anything docs to adjust parameters on clangd? 

I try to find features on clangd website.  I don't find any search file-related.

Thanks for your reading. any suggestion is welcome.

 

https://clangd.llvm.org/features

**Comment by HighCommander4** [comment, marked as answer]
Could you be more specific about which operation (e.g. Find References) you're performing which is slow?

Generally, I would expect using a remote index to address the performance impact of **building** the index locally, while the performance of operations that use the index should be largely unaffected.

**Reply by simon8233** [reply]
I try searching file name [CTRL + P on VSCode] is slow

search symbol [CTRL +T] is very fast 
I'm very grateful for your reply and community help

**Reply by HighCommander4** [reply]
> I try searching file name [CTRL + P on VSCode] is slow

This operation does not go through clangd at all -- it's done entirely on the client side in vscode. So, using a remote index has no impact on it.

**Reply by simon8233** [reply]
OK, Thanks for your help and answer. 
I will try other solutions to speed up search file operation.
